src/async_impl/client.rs

```diff
@@ -1340,6 +1343,16 @@ impl ClientBuilder {
             .insert(domain.to_string(), addrs.to_vec());
         self
     }
+
+    /// Override the DNS resolver implementation.
+    ///
+    /// Pass an `Arc` wrapping a trait object implementing `Resolve`.
+    /// Overrides for specific names passed to `resolve` and `resolve_to_addrs` will
+    /// still be applied on top of this resolver.
```

Comment:
This probably merits some discussion... 

- What if it were documented that the `resolve()` and `resolve_to_addrs()` were convenience methods that essentially called `dns_resolver()` for the user? That might make the implementation inside reqwest cleaner.
- But perhaps what you suggest here is easier for the user, and we should prefer that over easier for us?

Reply:
I think that documentation makes sense, I can add that. Not sure I follow how the implementation would change though - could you elaborate?

My intuition was:
* To retain backwards-compatibility, `resolve_to_addrs()` would need to keep track of a `HashMap` anyway, so setting `dns_resolver` in that method wouldn't be that helpful in terms of simplification.
* `resolve_to_addrs()` would also have to call logic to determine the default resolver in addition to `build`.

Reply:
Sorry, I didn't mean that we _should_ make a documentation or implementation change. Rather, I wanted to discuss what was best to do. But thinking about what you've said, I think the current solution here is likely best. Thanks :D